zyngui/zynthian_gui_control.py

```diff
@@ -330,21 +330,79 @@ def back_action(self):
 			return False
 
 
+	def layer_down(self):
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[0].nudge(-1)
+			except:
+				pass
+		return True
+
+
+	def layer_up(self):
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[0].nudge(1)
+			except:
+				pass
+		return True
+
+
+	def back_down(self):
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[1].nudge(-1)
+			except:
+				pass
+		return True
+
+
+	def back_up(self):
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[1].nudge(1)
+			except:
+				pass
+		return True
+
+
+	def snapshot_down(self):
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[2].nudge(-1)
+			except:
+				pass
+		return True
+
+
+	def snapshot_up(self):
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[2].nudge(1)
+			except:
+				pass
+		return True
+
+
 	def select_down(self):
-		i = self.index + 1
-		if i>=len(self.list_data):
-			i = 0
-		self.select(i)
-		self.click_listbox()
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[3].nudge(-1)
+			except:
+				pass
+		else:
+			super().select_down()
 		return True
 
 
 	def select_up(self):
-		i = self.index - 1
-		if i<0:
-			i = 0
-		self.select(i)
-		self.click_listbox()
+		if self.controllers_lock and self.mode=='control' and self.zcontrollers:
+			try:
+				self.zcontrollers[3].nudge(1)
+			except:
+				pass
+		else:
+			super().select_up()
```

Comment:
No my friend!!! You can't change the behaviour of things you don't understand like this!! Are you missing we have a new prototype called Z2? I will revert this change. Please, don't change things like this. I will send you a Z2 ASAP so you can test with it. Until then, please, think wise. There are lots of improvements & adjustments specifically done for the Z2.

Reply:
Also, this is very VX-centric. We are moving from this towards something more flexible. The classic "4 rotaries + 4 switches" still have to work, but it shouldn't be hardcoded anymore. I mean, we shouldn't hardcode switch/knobs actions like this.  Not in new code. In fact, we should refact old code to remove this.

Reply:
I am aware that these changes are likely to break stuff and have been methodically working through the modules to fix errors. Of course it is challenging to test the Z2 stuff but I had intended to look at its behaviour but tried to make the initial changes hardware agnostic.

VX-centric? Not sure what that means.